trajectory_tracker/src/trajectory_server.cpp

```diff
@@ -39,50 +39,48 @@
 #include <fstream>
 #include <string>
 
-#include <ros/ros.h>
+#include <rclcpp/rclcpp.hpp>
+#include <rclcpp/serialization.hpp>
+#include <rclcpp/serialized_message.hpp>
 
-#include <geometry_msgs/Twist.h>
-#include <interactive_markers/interactive_marker_server.h>
-#include <nav_msgs/Path.h>
+#include <geometry_msgs/msg/twist.hpp>
+#include <interactive_markers/interactive_marker_server.hpp>
+#include <nav_msgs/msg/path.hpp>
 #include <tf2_geometry_msgs/tf2_geometry_msgs.h>
-#include <trajectory_tracker_msgs/ChangePath.h>
-#include <trajectory_tracker_msgs/TrajectoryServerStatus.h>
-#include <visualization_msgs/InteractiveMarkerUpdate.h>
+#include <trajectory_tracker_msgs/srv/change_path.hpp>
+#include <trajectory_tracker_msgs/msg/trajectory_server_status.hpp>
+#include <visualization_msgs/msg/interactive_marker_update.hpp>
 
 #include <trajectory_tracker/filter.h>
 
-#include <neonavigation_common/compatibility.h>
-
-class ServerNode
+class ServerNode : public rclcpp::Node
 {
 public:
   ServerNode();
   ~ServerNode();
   void spin();
 
 private:
-  ros::NodeHandle nh_;
-  ros::NodeHandle pnh_;
-  ros::Publisher pub_path_;
-  ros::Publisher pub_status_;
-  ros::ServiceServer srv_change_path_;
-  interactive_markers::InteractiveMarkerServer srv_im_fb_;
+  rclcpp::Publisher<nav_msgs::msg::Path>::SharedPtr pub_path_;
+  rclcpp::Publisher<trajectory_tracker_msgs::msg::TrajectoryServerStatus>::SharedPtr pub_status_;
+  rclcpp::Service<trajectory_tracker_msgs::srv::ChangePath>::SharedPtr srv_change_path_;
+  std::shared_ptr<interactive_markers::InteractiveMarkerServer> srv_im_fb_;
 
-  nav_msgs::Path path_;
+  nav_msgs::msg::Path path_;
   std::string topic_path_;
-  trajectory_tracker_msgs::ChangePath::Request req_path_;
+  trajectory_tracker_msgs::srv::ChangePath::Request req_path_;
```

Comment:
preffer to use shared pointer for `req_path_`
```suggestion
  trajectory_tracker_msgs::srv::ChangePath::Request::SharedPtr req_path_;
```